crates/common/fork_choice/src/handlers.rs

```diff
@@ -110,37 +136,58 @@ pub fn on_attester_slashing(
 ) -> anyhow::Result<()> {
     let attestation_1 = attester_slashing.attestation_1;
     let attestation_2 = attester_slashing.attestation_2;
+
     ensure!(is_slashable_attestation_data(
         &attestation_1.data,
         &attestation_2.data
     ));
-    let state = &store.block_states[&store.justified_checkpoint.root];
+
+    let state = &store
+        .db
+        .beacon_state_provider()
+        .get(store.db.justified_checkpoint_provider().get()?.root)?
+        .ok_or_else(|| anyhow!("beacon state not found"))?;
+
     ensure!(state.is_valid_indexed_attestation(&attestation_1)?);
     ensure!(state.is_valid_indexed_attestation(&attestation_2)?);
 
     let attestation_1_indices = attestation_1
         .attesting_indices
         .into_iter()
         .collect::<HashSet<_>>();
+
     let attestation_2_indices = attestation_2
         .attesting_indices
         .into_iter()
         .collect::<HashSet<_>>();
+
+    let mut equivocating = match store.db.equivocating_indices_provider().get() {
+        Ok(set) => set,
+        Err(StoreError::FieldNotInitilized) => HashSet::default(),
+        Err(err) => return Err(err.into()),
+    };
```

Comment:
```suggestion
    let mut equivocating = store.db.equivocating_indices_provider().get()?;
```

can't we just do this?

Reply:
I switched this back because when i switched this to what you suggest i ran into this error
assertion `left == right` failed: Unexpected result on on_attester_slashing
  left: false
 right: true
 So I changed it back and the test is passing now